src/tests/functional/shared_test_classes/src/base/utils/calculate_thresholds.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/base/utils/calculate_thresholds.hpp"
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/model.hpp"
+
+#include "openvino/op/convert.hpp"
+
+#include "common_test_utils/ov_tensor_utils.hpp"
+
+namespace ov {
+namespace test {
+namespace utils {
+
+inline std::pair<double, double>
+calculate_thresholds_by_whole_model(const std::shared_ptr<ov::Model>& model) {
+    double max_abs_threshold = DISABLE_THRESHOLD, max_rel_threshold = DISABLE_THRESHOLD;
+
+    // check all operations except convert to generate correct values
+    for (const auto& op : model->get_ordered_ops()) {
+        if (std::dynamic_pointer_cast<ov::op::v0::Convert>(op)) {
+            continue;
+        }
+        // check the default threshold for operations
+        const auto it = custom_op_thresholds.find(op->get_type_info());
+        if (it != custom_op_thresholds.end()) {
+            if (it->second.abs_threshold > max_abs_threshold) {
+                max_abs_threshold = it->second.abs_threshold;
+            }
+            if (it->second.rel_threshold > max_rel_threshold) {
+                max_rel_threshold = it->second.rel_threshold;
+            }
+        }
+        for (const auto& in : op->inputs()) {
+            const auto elem_type = in.get_element_type();
+            const auto abs_value = ov::test::utils::tensor_comparation::calculate_default_abs_threshold(elem_type);
+            if (abs_value > max_abs_threshold) {
+                max_abs_threshold = abs_value;
+            }
+            const auto rel_value = ov::test::utils::tensor_comparation::calculate_default_rel_threshold(elem_type);
+            if (rel_value > max_rel_threshold) {
+                max_rel_threshold = rel_value;
+            }
+        }
```

Comment:
Looks like this part may be computed once out of `get_ordered_ops()`. And if `custom_op_thresholds` provides less strict threshold, it may be relaxed

Reply:
We calculate threshold only in case of not initialized thresholds